packages/slate-react/src/utils/dom.ts

```diff
@@ -233,3 +233,20 @@ export const getPlainText = (domNode: DOMNode) => {
 
   return text
 }
+
+const catchSlateFragment = /data-slate-fragment="(.+?)"/m
+export const getClipboardData = (dataTransfer: DataTransfer): DataTransfer => {
+  if (!dataTransfer.getData('application/x-slate-fragment')) {
+    const htmlData = dataTransfer.getData('text/html')
```

Comment:
Just so I'm clear, I think what this is doing is:

* On paste into a slate document, check to see if `application/x-slate-fragment` exists
* If true, return the dataTransfer and proceed
* If false, see if the html contains a `data-slate-fragment`.
* If that is true, convert that to a slate fragment and add it to the clipboardData and return that in place of the original DataTransfer and return. My question here is, this is now returning data only on `x-slate-fragment` instead of retaining whatever values were set for `text/html`, `text/plain`, etc., so this would likely break someone who is parsing both slate data and say plain text?
* If false, return the original dataTransfer and proceed

So I think we need to make sure that anything we return here is additive to the DataTransfer rather than just returning x-slate-fragment data. Unless I'm misunderstanding.

Reply:
Clearly I missed

```dataTransfer.types.forEach(type => {
        clipboardData.setData(type, dataTransfer.getData(type));
      });```

in my previous review. So I think we're good.